src/librustc_codegen_utils/codegen_backend.rs

```diff
@@ -43,11 +43,22 @@ pub trait CodegenBackend {
     /// # Panics
     ///
     /// Panics when the passed `Box<dyn Any>` was not returned by `codegen_backend`.
-    fn join_codegen_and_link(
+    fn join_codegen(
         &self,
         ongoing_codegen: Box<dyn Any>,
         sess: &Session,
         dep_graph: &DepGraph,
+    ) -> Result<Box<dyn Any>, ErrorReported>;
```

Comment:
Maybe return `Result<CodegenResults, ErrorReported>` instead? That would require merging `rustc_codegen_utils` into `rustc_codegen_ssa` though, so I am fine with leaving this as is too.

Reply:
Yup, I also thought about that route, but merging `rustc_codegen_utils` into `rustc_codegen_ssa` has other impacts; so I prefer to leave this as it.